File: scripts/export-tech-companies.py
import sqlite3
import csv
import os

TECH_ATS = [
  'ashbyhq', 
  'leverco', 
  'greenhouse', 
  'gem', 
  'jobvite',
  'rippling',
  'recruitee',
  'teamtailor'
]

def export_tech_companies():
    db_path = os.path.join(os.path.dirname(__file__), '..', 'scratch', 'OpenPostings', 'jobs.db')
    out_path = os.path.join(os.path.dirname(__file__), 'tech-companies-to-scrape.csv')
    
    conn = sqlite3.connect(db_path)
    cursor = conn.cursor()
    
    placeholders = ','.join(['?'] * len(TECH_ATS))
    query = f"""
        SELECT company_name, url_string, ATS_name 
        FROM companies 
        WHERE ATS_name IN ({placeholders})
        ORDER BY ATS_name, company_name
    """
    
    cursor.execute(query, TECH_ATS)
    results = cursor.fetchall()
    
    print(f"Found {len(results)} tech companies.")
    
    with open(out_path, 'w', newline='', encoding='utf-8') as f:
        writer = csv.writer(f)
        writer.writerow(['Company Name', 'Job Board URL', 'ATS Provider'])
        for row in results:
            writer.writerow(row)
            
    print(f"Exported to {out_path}")
    conn.close()

if __name__ == '__main__':
    export_tech_companies()

File: scripts/export-tech-companies.mjs
import sqlite3 from 'sqlite3';
import { open } from 'sqlite';
import { writeFileSync } from 'fs';
import { join } from 'path';

// Known ATS platforms that are overwhelmingly used by tech companies / startups
const TECH_ATS = [
  'ashbyhq', 
  'leverco', 
  'greenhouse', 
  'gem', 
  'jobvite',
  'rippling',
  'recruitee',
  'teamtailor'
];

async function exportTechCompanies() {
  const db = await open({
    filename: './scratch/OpenPostings/jobs.db',
    driver: sqlite3.Database
  });

  const placeholders = TECH_ATS.map(() => '?').join(',');
  const query = `
    SELECT company_name, url_string, ATS_name 
    FROM companies 
    WHERE ATS_name IN (${placeholders})
    ORDER BY ATS_name, company_name
  `;

  const companies = await db.all(query, TECH_ATS);
  
  console.log(`Found ${companies.length} tech companies.`);

  const headers = ['Company Name', 'Job Board URL', 'ATS Provider'];
  const lines = [headers.join(',')];

  for (const c of companies) {
    const name = `"${(c.company_name || '').replace(/"/g, '""')}"`;
    const url = `"${(c.url_string || '').replace(/"/g, '""')}"`;
    const ats = `"${(c.ATS_name || '').replace(/"/g, '""')}"`;
    lines.push(`${name},${url},${ats}`);
  }

  const outPath = join(process.cwd(), 'scripts', 'tech-companies-to-scrape.csv');
  writeFileSync(outPath, lines.join('\n'), 'utf-8');
  console.log(`Exported to ${outPath}`);
}

exportTechCompanies().catch(console.error);
